pgvectorscale/src/access_method/sbq.rs

```diff
@@ -527,43 +544,70 @@ impl<'a> SbqSpeedupStorage<'a> {
                     }
 
                     let rn_neighbor = unsafe {
-                        SbqNode::read(self.index, neighbor_index_pointer, &mut lsr.stats)
+                        SbqNode::read(
+                            self.index,
+                            neighbor_index_pointer,
+                            self.has_labels,
+                            &mut lsr.stats,
+                        )
                     };
+
                     let node_neighbor = rn_neighbor.get_archived_node();
                     let bq_vector = node_neighbor.get_bq_vector();
-                    let distance = lsr.sdm.as_ref().unwrap().calculate_bq_distance(
-                        bq_vector,
-                        gns,
-                        &mut lsr.stats,
-                    );
+                    let distance = lsr
```

Comment:
move distance to after check for labels. If labels do not match, no need to calculate distance